### PR TITLE
feat: Social media disruption detail

### PR DESCRIPTION
**Pull request by alainajaffer**
Please test this as various user roles.
I also added the back button as per DISRUPTION-226.

This should allow you to add social media posts (max 5) to the disruption detail page and then save them.

### CODE REVIEW

File: site/data/dynamo.ts [hunk 1]
```diff
@@ -571,6 +576,23 @@ export const getDisruptionById = async (disruptionId: string, id: string): Promi
...
         });
+
+        const pendingSocialMediaPosts = disruptionItems.filter(
+            (item) =>
+                ((item.SK as string).startsWith(`${disruptionId}#SOCIALMEDIAPOST`) &&
+                    (item.SK as string).endsWith("#PENDING")) ??
+                false,
+        );
+        pendingSocialMediaPosts.forEach((pendingSocialMediaPost) => {
+            const existingIndex = socialMediaPosts.findIndex(
```
Comment (alvak1) [inline site/data/dynamo.ts:587]: `consequences` needs to be replaced with `socialMediaPosts`

File: site/pages/disruption-detail/[disruptionId].page.tsx [hunk 2]
```diff
@@ -343,6 +352,148 @@ const DisruptionDetail = ({
...
+                                                            post.socialMediaPostIndex,
+                                                            true,
+                                                            true,
+                                                        ),
+                                                    ],
+                                                },
+                                                {
+                                                    header: "Status",
+                                                    cells: [post.status, ""],
+                                                },
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/pages/disruption-detail/[disruptionId].page.tsx:474]: solved in bug ticket

[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/pages/disruption-detail/[disruptionId].page.tsx:474]: solved in bug ticket

**Comment by alainajaffer** [comment]
> When a staff user has submitted a disruption for review - if the admin user rejects the disruption then error page is displayed instead of disruption status set to reject

system admins cannot reject disruptions

**Review by alvak1 — APPROVED** [review]
Approved

**Pull request merged**